maven-plugins/helidon-archetype-maven-plugin/src/main/java/io/helidon/build/maven/archetype/IntegrationTestMojo.java

```diff
@@ -548,6 +549,16 @@ private void invokePostArchetypeGenerationGoals(File basedir) throws IOException
         }
     }
 
+    private void ensureNoTemplates(File basedir) throws MojoExecutionException, IOException {
+        boolean isTemplatePresent = Files.walk(basedir.toPath())
+                .map(Path::getFileName)
+                .map(Path::toString)
+                .anyMatch(s -> s.endsWith(".mustache"));
```

Comment:
Template extension is arbitrary ; the archetypes declare templates with whatever extensions. Hard-coding `.mustache` here feels wrong.

How about using a set of globs for pattern matching and fail if there are matches (i.e. SourcePath).

Reply:
Can you tell me if it is what you expected, and some other template pattern than `.mustache` which I could add to the list ?

Reply:
This is not what I described. We should not hard-code `.mustache` because it's not hard-coding in the archetype engine. The archetypes decide what the filenames and extensions are. E.g. If one uses `.hbs` instead of `.mustache` then this mechanism wouldn't work.

You need to add a new parameter to the plugin that is a list of string where each item is a pattern. You will then match all patterns and fail accordingly.